File: src/main/java/com/example/temperaturerecords/repository/UserRepository.java
package com.example.temperaturerecords.repository;

import com.example.temperaturerecords.entities.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<UserEntity, Long> {
    Optional<UserEntity> findUserEntityByEmail(String email);
}

File: src/main/kotlin/com/example/temperaturerecords/repository/UserRepository.kt
package com.example.temperaturerecords.repository

import com.example.temperaturerecords.entities.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

interface UserRepository : JpaRepository<UserEntity, Long> {
    fun findUserEntityByUsername(username: String?): Optional<UserEntity?>
}
